test/mysql_handler/pkg/run_test_case.go

```diff
@@ -9,11 +10,26 @@ import (
 )
 
 
-func RunQuery(flags []string) (string, error) {
+func RunQuery(clientConfig test.ClientConfiguration, connectPort test.ConnectionPort) (string, error) {
 	args := []string{"-e", "select count(*) from testdb.test"}
 
-	for _, v := range flags {
-		args = append(args, v)
+	if clientConfig.SSL != nil && *clientConfig.SSL {
+		args = append(args, "--ssl", "--ssl-verify-server-cert=TRUE")
+	}
+	if clientConfig.Username != nil {
+		args = append(args, fmt.Sprintf("--user=%s", *clientConfig.Username))
+	}
+	if clientConfig.Password != nil {
+		args = append(args, fmt.Sprintf("--password=%s", *clientConfig.Password))
+	}
+	switch connectPort.ListenerType {
```

Comment:
The need for this switch statement means we still haven't found the concepts with the right "natural joints".  I'll return to this tomorrow and elaborate.